res/cmake/func.cmake

```diff
@@ -0,0 +1,72 @@
+
```

Comment:
does it make sense actually to have a "function" file where all functions are ? wouldn't it be easier to find these functions with the config they are associated with ? like those would go in test cmake ?

Reply:
if they all have a file they belong to it makes sense
is there any orphans?